python-frontend/src/main/java/org/sonar/python/types/RuntimeType.java

```diff
@@ -52,9 +55,27 @@ public boolean isIdentityComparableWith(InferredType other) {
     if (other instanceof UnionType) {
       return other.isIdentityComparableWith(this);
     }
+    if (isComparingTypeWithMetaclass(other)) {
+      return true;
+    }
     return this.equals(other);
   }
 
+  private boolean isComparingTypeWithMetaclass(InferredType other) {
+    if (other instanceof RuntimeType otherRuntimeType) {
+      boolean hasOtherMetaClass = hasMetaclassInHierarchy(otherRuntimeType);
+      boolean hasThisMetaClass = hasMetaclassInHierarchy(this);
+      return (TYPE_CLASS_NAME.equals(getTypeClass().name()) && hasOtherMetaClass)
```

Comment:
I guess there could be just a comparison with the `org.sonar.python.types.InferredTypes#TYPE` constant.